fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4

```diff
@@ -331,6 +331,15 @@ namedExpressionSeq
 
 expression
     : booleanExpression
+    | lambdaExpression
+    ;
+
+lambdaExpression
+    : args+=errorCapturingIdentifier ARROW body=booleanExpression
```

Comment:
```suggestion
    : args+=errorCapturingIdentifier ARROW body=expression
```

Reply:
body should be boolean expression, it can't be lambdaExpreesion

Reply:
why not must be bool,  seems could:  x->abs(x)